android/app/src/main/java/com/zulipmobile/notifications/NotificationHelper.kt

```diff
@@ -65,19 +65,31 @@ fun sizedURL(context: Context, url: URL, dpSize: Float): URL {
     return URL(url, "?$query")
 }
 
-private fun extractName(key: String): String {
-    return key.split(":")[0]
+fun buildNotificationContent(conversations: ByConversationMap, inboxStyle: Notification.InboxStyle, mContext: Context) {
+    for (messages in conversations.values) {
+        val messagesByNameMap = buildMessagesByNameMap(messages)
+        for ((name, messagesForName) in messagesByNameMap) {
```

Comment:
This doesn't accomplish the grouping requested in the issue #3591; in the example there, instead of
> **2 messages in 1 conversation**
> **A, B**: I'm B

it would show, I think, an arbitrary one of
> **2 messages in 1 conversation**
> **A**: I'm A
> **B**: I'm B

or
> **2 messages in 1 conversation**
> **B**: I'm B
> **A**: I'm A

(independent of the order the messages were sent/received in), and if there are several conversations, the lines for them would be concatenated with no indication of which went with which.

I think that's why you marked the PR as WIP, though.

Reply:
Yes, I am going to do this in a different commit.